tls/s2n_record_write.c

```diff
@@ -219,6 +219,13 @@ int s2n_record_writev(struct s2n_connection *conn, uint8_t content_type, const s
     uint16_t block_size = 0;
     uint8_t aad_iv[S2N_TLS_MAX_IV_LEN] = { 0 };
 
+    if (conn->send_mode == S2N_DEFAULT_SEND) {
+        /* If the user has not specified a send buffer size then the TLS records
+         * are flushed immediately. This implies a logic error if the stuffer was
+         * not reset in this mode. */
+        POSIX_ENSURE(!s2n_stuffer_data_available(&conn->out), S2N_ERR_RECORD_STUFFER_NEEDS_DRAINING);
+    }
```

Comment:
Why would this behavior change for S2N_MULTI_RECORD_SEND mode? If we're starting on a new write, shouldn't flush have completed the last write, regardless of how many records that involves?

Reply:
`s2n_record_writev` will only write one record at a time, so we have to invoke it multiple times with `S2N_MULTI_RECORD_SEND`. 

Changing this function to write multiple records at once doesn't seem necessary for this feature, but it does make this check fail in `S2N_MULTI_RECORD_SEND` mode.